tests/system/action/poll/test_vote.py

```diff
@@ -577,3 +596,1207 @@ def test_default_vote_weight(self) -> None:
+
+    def test_vote(self) -> None:
+        self.start_poll()
+        response = self.request(
+            "poll.vote",
+            {"value": {"1": 1, "2": 0}, "id": 1, "user_id": 1},
+        )
+        self.assert_status_code(response, 200)
+        self.assert_model_exists("vote/1")
+        self.assert_model_not_exists("vote/2")
+        poll = self.get_model("poll/1")
+        self.assertEqual(poll.get("votesvalid"), "1.000000")
+        self.assertEqual(poll.get("votesinvalid"), "0.000000")
+        self.assertEqual(poll.get("votescast"), "1.000000")
+        self.assertEqual(poll.get("state"), Poll.STATE_STARTED)
+        self.assertIn(1, poll.get("voted_ids", []))
+        option1 = self.get_model("option/1")
+        option2 = self.get_model("option/2")
+        self.assertEqual(option1.get("yes"), "1.000000")
+        self.assertEqual(option1.get("no"), "0.000000")
+        self.assertEqual(option1.get("abstain"), "0.000000")
+        self.assertEqual(option2.get("yes"), "0.000000")
+        self.assertEqual(option2.get("no"), "0.000000")
+        self.assertEqual(option2.get("abstain"), "0.000000")
+
+    def test_change_vote(self) -> None:
+        self.start_poll()
+        response = self.request(
+            "poll.vote",
+            {"value": {"1": 1, "2": 0}, "id": 1, "user_id": 1},
+        )
+        response = self.request(
+            "poll.vote",
+            {"value": {"1": 0, "2": 1}, "id": 1, "user_id": 1},
+        )
+        self.assert_status_code(response, 400)
+        option1 = self.get_model("option/1")
+        option2 = self.get_model("option/2")
+        self.assertEqual(option1.get("yes"), "1.000000")
+        self.assertEqual(option1.get("no"), "0.000000")
+        self.assertEqual(option1.get("abstain"), "0.000000")
+        self.assertEqual(option2.get("yes"), "0.000000")
+        self.assertEqual(option2.get("no"), "0.000000")
+        self.assertEqual(option2.get("abstain"), "0.000000")
+
+    def test_global_yes(self) -> None:
+        self.start_poll()
+        self.create_model(
+            "option/11", {"meeting_id": 113, "used_as_global_option_in_poll_id": 1}
+        )
+        self.update_model("poll/1", {"global_option_id": 11})
+        response = self.request("poll.vote", {"value": "Y", "id": 1, "user_id": 1})
+        self.assert_status_code(response, 200)
+        option = self.get_model("option/11")
+        self.assertEqual(option.get("yes"), "1.000000")
+        self.assertEqual(option.get("no"), "0.000000")
+        self.assertEqual(option.get("abstain"), "0.000000")
+
+    def test_global_yes_forbidden(self) -> None:
+        self.update_model("poll/1", dict(global_yes=False))
+        self.start_poll()
+        response = self.request("poll.vote", {"value": "Y", "id": 1, "user_id": 1})
+        self.assert_status_code(response, 400)
+        self.assert_model_not_exists("vote/1")
+
+    def test_global_no(self) -> None:
+        self.start_poll()
+        self.create_model(
```

Comment:
maybe create a function `create_global_option`

Reply:
this can also be done in the setup - this options is always present in the real scenario.

Reply:
Always add the global option now.